internal/querycoord/index_checker.go

```diff
@@ -136,6 +120,88 @@ func (ic *IndexChecker) reloadFromKV() error {
 	return nil
 }
 
+func (ic *IndexChecker) handleHandoffRequest(info *querypb.SegmentInfo) error {
+	// allocate a taskID for locking
+	taskID, err := ic.idAllocator()
+	if err != nil {
+		log.Warn("handleHandoffRequest: allocate taskID failed", zap.Error(err))
+		return err
+	}
+
+	// acquire reference lock and defer unlock
+	if err = ic.broker.acquireSegmentsReferLock(ic.ctx, taskID, []UniqueID{info.SegmentID}); err != nil {
+		log.Warn("acquire segment reference lock failed", zap.Int64("taskID", taskID), zap.Int64("segmentID", info.SegmentID))
+		return err
+	}
+	defer func() {
+		if err := ic.broker.releaseSegmentReferLock(ic.ctx, taskID, []UniqueID{info.SegmentID}); err != nil {
+			log.Warn("release segment reference lock failed", zap.Int64("taskID", taskID), zap.Int64("segmentID", info.SegmentID))
+			panic(err)
+		}
+	}()
+
+	collectionInfo, err := ic.meta.getCollectionInfoByID(info.CollectionID)
+	if err != nil {
+		log.Warn("handleHandoffRequest: handoffTask failed", zap.Int64("taskID", taskID), zap.Int64("segmentID", info.SegmentID), zap.Error(err))
+		return err
+	}
+
+	// wait until index(if any) is ready
+	indexInfo, err := ic.broker.getIndexInfo(ic.ctx, info.CollectionID, info.SegmentID, collectionInfo.Schema)
+	for err != nil {
+		time.Sleep(20 * time.Millisecond)
```

Comment:
Index building is relatively slow, this sleep time can  be longer. otherwise a large number of logs may be generated

Reply:
Agree with @xige-16. 
Best implementation here is that the sleep time could be configurable with a minimal interval limitation.
Also, select with ticker and context will look better here.

Reply:
Change it to 1 second.